Crust/Mapper/MappingOperator.swift

```diff
@@ -246,91 +263,230 @@ private func mapFromJson<T, U: Mapping>(_ json: JSONValue, toField field: inout
+    _ json: JSONValue,
+    fromField field: V,
+    viaKey key: Keypath,
+    mapping: U)
+    throws -> JSONValue
+    where U.MappedObject == T, V.Iterator.Element == T, U.SequenceKind.Iterator.Element == U.MappedObject {
+        
+        var json = json
+        
+        let results = try field.map {
+            try Mapper<U>().mapFromObjectToJSON($0, mapping: mapping)
+        }
+        json[key] = .array(results)
+        
+        return json
+}
+
+@discardableResult
+public func mapFromJson<T, U: Mapping, C: MappingContext>(toCollection field: inout U.SequenceKind, map:(key: Spec<U>, context: C)) -> C where U.MappedObject == T, U.SequenceKind: RangeReplaceableCollection, U.SequenceKind.Iterator.Element == U.MappedObject, T: Equatable {
     
-    let mapping = map.key.mapping
     do {
         switch map.context.dir {
         case .toJSON:
             let json = map.context.json
-            try map.context.json = mapToJson(json, fromField: field, viaKey: map.key, mapping: mapping)
-        case .fromJSON:
+            try map.context.json = mapToJson(json, fromField: field, viaKey: map.key, mapping: map.key.mapping)
             
-            let mapper = { (json: JSONValue, field: inout V) in
-                let allowDupes = map.key.options.contains(.AllowDuplicatesInCollection)
-                try mapFromJson(json, toField: &field, mapping: mapping, context: map.context, allowDuplicates: allowDupes)
+        case .fromJSON:
+            let fieldCopy = field
+            let (newObjects, _) = try mapFromJsonToSequence(map: map) {
+                fieldCopy.contains($0)
             }
             
-            let json = map.context.json
-            let baseJSON = json[map.key]
-            if case .some(.array(let arr)) = baseJSON, map.key.keyPath == "", arr.count == 0 {
-                try mapper(json, &field)
-            }
-            else if let baseJSON = baseJSON {
-                try mapper(baseJSON, &field)
-            }
-            else {
-                let userInfo = [ NSLocalizedFailureReasonErrorKey : "JSON at key path \(map.key) does not exist to map from" ]
-                throw NSError(domain: CrustMappingDomain, code: 0, userInfo: userInfo)
+            switch map.key.collectionUpdatePolicy.insert {
+            case .append:
+                field.append(contentsOf: newObjects)
+                
+            case .replace(delete: let deletionBlock):
+                var orphans = field
+                field = U.SequenceKind(newObjects)
+                
+                if let deletion = deletionBlock {
+                    field.forEach {
+                        if let index = orphans.index(of: $0) {
+                            orphans.remove(at: index)
+                        }
+                    }
+                    
+                    try deletion(orphans).forEach {
+                        try map.key.mapping.delete(obj: $0)
+                    }
```

Comment:
Where does the replacement happen? I only see delete in this case. Is it handled afterwards somewhere?